File: leetcode/239.sliding-window-maximum/go/maxslidingwindow_test.go
package maxslidingwindow

import (
	"reflect"
	"testing"
)

type testcase struct {
	nums []int
	k    int
	want []int
}

func TestMaxSlidingWindow(t *testing.T) {
	tcs := []testcase{
		{nums: []int{1, 3, -1, -3, 5, 3, 6, 7}, k: 3, want: []int{3, 3, 5, 5, 6, 7}},
	}

	for _, tc := range tcs {
		t.Run("maxslidingwindow", func(t *testing.T) {
			got := maxSlidingWindow(tc.nums, tc.k)
			if !reflect.DeepEqual(got, tc.want) {
				t.Fatalf("maxslidingwindow, want: %v, got: %v", tc.want, got)
			}
		})
	}
}

File: leetcode/239.sliding-window-maximum/cpp/maxslidingwindow_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "maxslidingwindow.cpp"

struct testCase
{
    std::vector<int> nums;
    int k;
    std::vector<int> want;
};

TEST_CASE("Max Sliding Window")
{
    std::vector<testCase>
        tcs{
            {{1, 3, -1, -3, 5, 3, 6, 7}, 3, {3, 3, 5, 5, 6, 7}},
        };

    Solution s;

    for (auto tc : tcs)
    {
        auto got = s.maxSlidingWindow(tc.nums, tc.k);
        CHECK(got == tc.want);
    }
}
